app/src/androidTestKiwix/java/org/kiwix/kiwixmobile/tests/DownloadTest.java

```diff
@@ -38,104 +40,97 @@
+import org.kiwix.kiwixmobile.utils.SplashActivity;
 
 import java.util.concurrent.TimeUnit;
 
 import static android.support.test.espresso.Espresso.onData;
 import static android.support.test.espresso.Espresso.onView;
 import static android.support.test.espresso.action.ViewActions.click;
-import static android.support.test.espresso.action.ViewActions.longClick;
-import static android.support.test.espresso.action.ViewActions.swipeDown;
 import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
 import static android.support.test.espresso.matcher.ViewMatchers.withId;
 import static android.support.test.espresso.matcher.ViewMatchers.withParent;
-import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions.assertDisplayed;
+import static com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn;
+import static com.schibsted.spain.barista.interaction.BaristaDialogInteractions.clickDialogPositiveButton;
+import static com.schibsted.spain.barista.interaction.BaristaSwipeRefreshInteractions.refresh;
 import static org.hamcrest.Matchers.allOf;
+import static org.kiwix.kiwixmobile.testutils.TestUtils.allowPermissionsIfNeeded;
+import static org.kiwix.kiwixmobile.testutils.TestUtils.getResourceString;
 import static org.kiwix.kiwixmobile.testutils.TestUtils.withContent;
+import static org.kiwix.kiwixmobile.testutils.TestUtils.getCurrentActivity;
+import static org.kiwix.kiwixmobile.utils.StandardActions.deleteZimIfExists;
+import static org.kiwix.kiwixmobile.utils.StandardActions.enterHelp;
 
 @LargeTest
 @RunWith(AndroidJUnit4.class)
 public class DownloadTest {
 
+  public static final String KIWIX_DOWNLOAD_TEST = "kiwixDownloadTest";
   @Rule
-  public ActivityTestRule<ZimManageActivity> mActivityTestRule = new ActivityTestRule<>(
-      ZimManageActivity.class);
+  public ActivityTestRule<SplashActivity> mActivityTestRule = new ActivityTestRule<>(SplashActivity.class);
   @Rule
   public GrantPermissionRule readPermissionRule = GrantPermissionRule.grant(Manifest.permission.READ_EXTERNAL_STORAGE);
   @Rule
   public GrantPermissionRule writePermissionRule = GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE);
 
   @BeforeClass
   public static void beforeClass() {
-    IdlingPolicies.setMasterPolicyTimeout(350, TimeUnit.SECONDS);
-    IdlingPolicies.setIdlingResourceTimeout(350, TimeUnit.SECONDS);
+    IdlingPolicies.setMasterPolicyTimeout(180, TimeUnit.SECONDS);
+    IdlingPolicies.setIdlingResourceTimeout(180, TimeUnit.SECONDS);
     Espresso.registerIdlingResources(KiwixIdlingResource.getInstance());
   }
 
   @Before
   public void setUp() {
-
   }
 
   @Test
-  public void downloadTest() {
-    ViewInteraction appCompatTextView = onView(
-        allOf(withText("Device"), isDisplayed()));
-    appCompatTextView.perform(click());
+  public void downloadTest() throws Throwable {
```

Comment:
Throwable is extremely broad, can you practically narrow the set of exceptions this throws?